linera-execution/src/committee.rs

```diff
@@ -9,18 +9,81 @@ use linera_base::{
     data_types::ArithmeticError,
 };
 use serde::{Deserialize, Serialize};
-use std::{borrow::Cow, collections::BTreeMap};
+use std::{borrow::Cow, collections::BTreeMap, str::FromStr};
 
 /// A number identifying the configuration of the chain (aka the committee).
-#[derive(
-    Eq, PartialEq, Ord, PartialOrd, Copy, Clone, Hash, Default, Debug, Serialize, Deserialize,
-)]
+#[derive(Eq, PartialEq, Ord, PartialOrd, Copy, Clone, Hash, Default, Debug)]
 pub struct Epoch(pub u64);
 
+impl Serialize for Epoch {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::ser::Serializer,
+    {
+        if serializer.is_human_readable() {
+            serializer.serialize_str(&self.0.to_string())
```

Comment:
Based on my tests, we should be able to serialize and deserialize as `u64` here. It won't change anything for JSON keys (still strings) but will be cleaner in other cases.

Reply:
The only issue is that if people read it as a JavaScript number, it can get rounded, since the mantissa has only 52 bits.

But realistically, epoch numbers shouldn't get that high anyway. Maybe we should make them `u32` instead of `u64` in the first place?

Reply:
> Based on my tests, we should be able to serialize and deserialize as `u64` here. It won't change anything for JSON keys (still strings) but will be cleaner in other cases.

The JSON serialization works but not the GraphQL one, as shown [there](https://github.com/async-graphql/async-graphql/blob/master/value/src/serializer.rs#L439-L640), the GraphQL serialization only allows `string` in ma keys.
You can test it by launching `run_local.sh`, the node service and requesting
```
query {
  blocks(chainId: "e476187f6ddfeb9d588c7b45d3df334d5501d6499b3f9ad5595cae86cce16a65") {
    value {
      executedBlock {
        block {
          height
          operations
        }
      }
    }
  }
}
```
on GraphiQL.
The `openChain` operation of block 0 will be serialized as `null`.
You can also use the `async_graphql::to_value` function that will answer the `"Key must be a string"` error.